### PR TITLE
Approve reject super status admin

### PR DESCRIPTION
**Pull request by dupreesi**
relates to #77, #76, #7 

ADMIN USER CAN APPROVE / REJECT SUPER USER AND USER VERIFICATIONS
- [x] creates feature to approve or reject users awaiting verification and/or awaiting super user status
- [x] adds queries to approve, reject super users / users
- [x] adds handler that deals with onClick requests to approve/ reject
- [x] restructures Admin/User component structure to clean up User/index.js router
- [x] filters user data according to various sections
- [x] adds custom hook: hooks/postPatchPut.js to be used for api calls (example usage in Admin/Users/AllUsers.js as patch request)

to test rebuild db, go to Admin/Users/AllUsers.js and modify 
```
const admin = 'hardcoded admin user id'
```
then go to admin control panel and hit users...

### CODE REVIEW

File: src/usePostPatchPut.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { useState, useCallback } from 'react';
```
Comment (arrested-developer) [inline src/usePostPatchPut.js]: Is this file a duplicate?

Comment (amusameh) [reply src/usePostPatchPut.js]: @arrested-developer  I added usePost.js hook but I didn't use it in the end. so it's the that is redundant. I forgot to delete it

File: src/lambda/database/queries/user/index.js [hunk 2]
```diff
@@ -1,3 +1,45 @@
...
+// gets one user
+export const getUser = id => {
+  return User.findById(id);
+};
+
+// upates awaitingSuperUser status if rejected
+export const rejectSuperUser = id =>
+  User.findOneAndUpdate(
+    { _id: id },
+    { status: status.VERIFIED },
```
Comment (amusameh) [inline src/lambda/database/queries/user/index.js]: This should be rejected not verified
```suggestion
    { status: status.REJECTED},
```

File: src/lambda/manageUserStatus.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+} from './database/queries/user';
+import { status } from '../constants/users';
+
+export async function handler(event, context) {
+  // eslint-disable-next-line no-param-reassign
+  context.callbackWaitsForEmptyEventLoop = false;
+
+  try {
+    await connectToDatabase();
+    const { user, admin, action, userStatus } = JSON.parse(event.body);
```
Comment (amusameh) [inline src/lambda/manageUserStatus.js]: I think you should validate the data here. check if it exists and it's what's expected

File: src/lambda/manageUserStatus.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    let update;
+    let msg;
+    const adminExists = await getUser(admin);
+    // check if admin ID Error
+    if (!adminExists) {
```
Comment (amusameh) [inline src/lambda/manageUserStatus.js]: You should check for the user role to be sure they are admin. now any user will be treated as an admin

File: src/lambda/database/queries/user/index.js (hunk 2 above)
Comment (dupreesi) [reply src/lambda/database/queries/user/index.js]: @amusameh given the current model when a user with status awaitingSuper is getting rejected this user will still be a verified user.

File: src/lambda/manageUserStatus.js (hunk 4 above)
Comment (dupreesi) [reply src/lambda/manageUserStatus.js]: @amusameh good point but I'd like to wait for the auth0 PR to be merged in and then create a reusable role check strategy if thats alright. Not only admin users but super users will be able to use this handler so would be good to set up middlewares to check for that throughout the app

File: src/lambda/manageUserStatus.js (hunk 3 above)
Comment (dupreesi) [reply src/lambda/manageUserStatus.js]: good point I'll do that

File: src/lambda/database/queries/user/index.js (hunk 2 above)
Comment (amusameh) [reply src/lambda/database/queries/user/index.js]: @dupreesi yeah that make sense. I missed that.

File: src/lambda/manageUserStatus.js (hunk 4 above)
Comment (amusameh) [reply src/lambda/manageUserStatus.js]: @dupreesi good idea I agree

**Pull request merged**